frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx

```diff
@@ -0,0 +1,151 @@
+
+const logExplorerRoute = '/logs/logs-explorer';
+
+const queryRangeURL = 'http://localhost/api/v3/query_range';
+
+const lodsQueryServerRequest = (): void =>
+	server.use(
+		rest.post(queryRangeURL, (req, res, ctx) =>
+			res(ctx.status(200), ctx.json(logsQueryRangeSuccessResponse)),
+		),
+	);
+
+// mocking the graph components in this test as this should be handled separately
+jest.mock(
+	'container/TimeSeriesView/TimeSeriesView',
+	() =>
+		// eslint-disable-next-line func-names, @typescript-eslint/explicit-function-return-type, react/display-name
+		function () {
+			return <div>Time Series Chart</div>;
+		},
+);
+jest.mock(
+	'container/LogsExplorerChart',
+	() =>
+		// eslint-disable-next-line func-names, @typescript-eslint/explicit-function-return-type, react/display-name
+		function () {
+			return <div>Histogram Chart</div>;
+		},
+);
+
+jest.mock('constants/panelTypes', () => ({
+	AVAILABLE_EXPORT_PANEL_TYPES: ['graph', 'table'],
+}));
+
+jest.mock('d3-interpolate', () => ({
+	interpolate: jest.fn(),
+}));
+
+jest.mock('hooks/queryBuilder/useGetExplorerQueryRange', () => ({
+	__esModule: true,
+	useGetExplorerQueryRange: jest.fn(),
+}));
+
+// Set up the specific behavior for useGetExplorerQueryRange in individual test cases
+beforeEach(() => {
+	(useGetExplorerQueryRange as jest.Mock).mockReturnValue({
+		data: { payload: logsQueryRangeSuccessNewFormatResponse },
+	});
+});
+
+const renderer = (): RenderResult =>
+	render(
+		<MemoryRouter initialEntries={[logExplorerRoute]}>
+			<Provider store={store}>
+				<I18nextProvider i18n={i18n}>
+					<MockQueryClientProvider>
+						<QueryBuilderProvider>
+							<VirtuosoMockContext.Provider
+								value={{ viewportHeight: 300, itemHeight: 100 }}
+							>
+								<LogsExplorerViews selectedView={SELECTED_VIEWS.SEARCH} showHistogram />
+							</VirtuosoMockContext.Provider>
+						</QueryBuilderProvider>
+					</MockQueryClientProvider>
+				</I18nextProvider>
+			</Provider>
+		</MemoryRouter>,
+	);
+
+describe('LogsExplorerViews -', () => {
+	it('render correctly with props - list and table', async () => {
+		lodsQueryServerRequest();
+		const { queryByText, queryByTestId } = renderer();
+
+		expect(queryByTestId('periscope-btn')).toBeInTheDocument();
+		await userEvent.click(queryByTestId('periscope-btn') as HTMLElement);
+
+		expect(document.querySelector('.menu-container')).toBeInTheDocument();
+
+		const menuItems = document.querySelectorAll('.menu-items .item');
+		expect(menuItems.length).toBe(3);
+
+		// switch to table view
+		// eslint-disable-next-line sonarjs/no-duplicate-string
+		await userEvent.click(queryByTestId('table-view') as HTMLElement);
+
+		expect(
+			queryByText(
+				'{"container_id":"container_id","container_name":"container_name","driver":"driver","eta":"2m0s","location":"frontend","log_level":"INFO","message":"Dispatch successful","service":"frontend","span_id":"span_id","trace_id":"span_id"}',
+			),
+		);
```

Comment:
expect(queryByText("fbhf")) ? assertion seems missing here

Reply:
Thanks, these got cleared out somehow is batch editing or something, fixed everywhere